### DISCUSSION TITLE
Get scroll option in router.events when a route changed

### DISCUSSION THREAD

**Discussion by dnbastos** [vercel/next.js#32225] [Ideas]
### Describe the feature you'd like to request

My application has a layout that is a scrollable box, a div with a limited size that makes the main scroll not on the main page element. Because of this the option to scroll to the top of the page when changing route does not work.

This can be resolved in a way that has already been mentioned in other issues:

```js
  Router.events.on('routeChangeComplete', () => { containerRef.scrollTo(0, 0); }); 
```

The problem with this solution is that there are occasions when I want to turn off this scroll behavior. This is a real problem, so much so that to solve it, the API  of `router.push` and `Link` have an option to disable scroll by passing `{ scroll: false }`. That would solve my problem, but it doesn't work in my case.

This could be resolved if the route events (`Router.events`) had the scroll option that was used in the redirect. So I could solve the problem like this:

```js
  Router.events.on('routeChangeComplete', (url, { scroll }) => { 
    if (scroll) {    
      containerRef.scrollTo(0, 0);     
    }
  });
```

### Describe the solution you'd like

The router events would have the scroll option:
`routeChangeComplete(url, { shallow, scroll })`

### Describe alternatives you've considered

For now I'm leaving the scroll enabled for all cases.

**Comment by michalkvasnicak** [comment]
This would be really helpful
